flink-formats/flink-avro-glue-schema-registry/src/main/java/org/apache/flink/formats/avro/glue/schema/registry/GlueSchemaRegistryAvroSerializationSchema.java

```diff
@@ -0,0 +1,127 @@
+ */
+public class GlueSchemaRegistryAvroSerializationSchema<T>
+        extends RegistryAvroSerializationSchema<T> {
+    /**
+     * Creates an Avro serialization schema.
+     *
+     * @param recordClazz class to serialize. Should be one of: {@link SpecificRecord}, {@link
+     *     GenericRecord}.
+     * @param reader reader's Avro schema. Should be provided if recordClazz is {@link
+     *     GenericRecord}
+     * @param schemaCoderProvider schema coder provider which reads writer schema from AWS Glue
+     *     Schema Registry
+     */
+    private GlueSchemaRegistryAvroSerializationSchema(
+            Class<T> recordClazz,
+            @Nullable Schema reader,
+            SchemaCoder.SchemaCoderProvider schemaCoderProvider) {
+        super(recordClazz, reader, schemaCoderProvider);
+    }
+
+    @VisibleForTesting
+    protected GlueSchemaRegistryAvroSerializationSchema(
+            Class<T> recordClazz, @Nullable Schema reader, SchemaCoder schemaCoder) {
+        // Pass null schema coder provider
+        super(recordClazz, reader, null);
+        this.schemaCoder = schemaCoder;
+    }
+
+    /**
+     * Creates {@link GlueSchemaRegistryAvroSerializationSchema} that serializes {@link
+     * GenericRecord} using provided schema.
+     *
+     * @param schema the schema that will be used for serialization
+     * @param transportName topic name or stream name etc.
+     * @param configs configuration map of AWS Glue Schema Registry
+     * @return serialized record in form of byte array
+     */
+    public static GlueSchemaRegistryAvroSerializationSchema<GenericRecord> forGeneric(
+            Schema schema, String transportName, Map<String, Object> configs) {
+        return new GlueSchemaRegistryAvroSerializationSchema<>(
+                GenericRecord.class,
+                schema,
+                new GlueSchemaRegistryAvroSchemaCoderProvider(transportName, configs));
+    }
+
+    /**
+     * Creates {@link GlueSchemaRegistryAvroSerializationSchema} that serializes {@link
+     * SpecificRecord} using provided schema.
+     *
+     * @param clazz the type to be serialized
+     * @param transportName topic name or stream name etc.
+     * @param configs configuration map of Amazon Schema Registry
+     * @return serialized record in form of byte array
+     */
+    public static <T extends SpecificRecord>
+            GlueSchemaRegistryAvroSerializationSchema<T> forSpecific(
+                    Class<T> clazz, String transportName, Map<String, Object> configs) {
+        return new GlueSchemaRegistryAvroSerializationSchema<>(
+                clazz, null, new GlueSchemaRegistryAvroSchemaCoderProvider(transportName, configs));
+    }
+
+    /**
+     * Serializes the incoming element to a byte array containing bytes of AWS Glue Schema registry
+     * information.
+     *
+     * @param object The incoming element to be serialized
+     * @return The serialized bytes.
+     */
+    @SneakyThrows
```

Comment:
See parent implementation. Wrap or declare the exception rather than sneaky throws.

Reply:
Fixed